library/pod-security-policy/seccomp/template.yaml

```diff
@@ -35,52 +35,220 @@ spec:
-            value := metadata.annotations[key]
-            startswith(key, "container.seccomp.security.alpha.kubernetes.io/")
-            [prefix, name] := split(key, "/")
-            name == container.name
-            profile = value
+        # Localhost files from annotation scheme
+        get_annotation_localhost_files[file] {
+        	profile := input.parameters.allowedProfiles[_]
+        	startswith(profile, "localhost/")
+        	file := split(profile, "/")[1]
         }
 
-        input_containers[c] {
-            c := input.review.object.spec.containers[_]
+        # The profiles explicitly in the list
+        get_allowed_profiles[allowed] {
+        	allowed := input.parameters.allowedProfiles[_]
         }
-        input_containers[c] {
-            c := input.review.object.spec.initContainers[_]
+
+        # The simply translated profiles
+        get_allowed_profiles[allowed] {
+        	profile := input.parameters.allowedProfiles[_]
+        	not startswith(lower(profile), "localhost")
+        	allowed := naming_translation[profile][_]
+        }
+
+        # Seccomp Localhost to annotation translation
+        get_allowed_profiles[allowed] {
+        	profile := input.parameters.allowedProfiles[_]
+        	profile == "Localhost"
+        	file := object.get(input.parameters, "allowedLocalhostFiles", [])[_]
+        	allowed := sprintf("%v/%v", [naming_translation[profile][_], file])
+        }
+
+        # Annotation localhost to Seccomp translation
+        get_allowed_profiles[allowed] {
+        	profile := input.parameters.allowedProfiles[_]
+        	startswith(profile, "localhost")
+        	allowed := naming_translation.localhost[_]
+        }
+
+        # Container profile as defined in pod annotation
+        get_profile(container) = {"profile": profile, "file": "", "location": location} {
+        	not has_securitycontext_container(container)
+        	not has_annotation(get_container_annotation_key(container.name))
+        	not has_securitycontext_pod
+        	profile := input.review.object.metadata.annotations[pod_annotation_key]
+        	location := sprintf("annotation %v", [pod_annotation_key])
+        }
+
+        # Container profile as defined in container annotation
+        get_profile(container) = {"profile": profile, "file": "", "location": location} {
+        	not has_securitycontext_container(container)
+        	container_annotation := get_container_annotation_key(container.name)
+        	has_annotation(container_annotation)
+        	profile := input.review.object.metadata.annotations[container_annotation]
+        	location := sprintf("annotation %v", [container_annotation])
+        }
+
+        # Container profile as defined in pods securityContext
+        get_profile(container) = {"profile": profile, "file": file, "location": location} {
+        	not has_securitycontext_container(container)
+        	not has_annotation(get_container_annotation_key(container.name))
```

Comment:
probably want to remove the container annotation check here for the same reason as why we want to add a pod SC check for the container annotation case above.

Reply:
@sozercan was this comment addressed?

Reply:
oops, thought this was merged into main for some reason

Reply:
This has been done, thanks